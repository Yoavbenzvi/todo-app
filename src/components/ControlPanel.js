import React from 'react';

const ControlPanel = ({ showAll, toggleShow, clearDisplay }) => {
	const active = 'text-xs mr-3 hover:underline text-gray-500 focus:outline-none text-green-500 font-bold'
	const disabled = 'text-xs mr-3 hover:underline text-gray-500 focus:outline-none'

	return(
		<div className='flex py-4 border-t border-gray-900 justify-between'>
			<div>
				<button
					id='all'
					className={showAll ? active : disabled}
					onClick={() => toggleShow(true)}
				>
					ALL
				</button>
				<button
					id='active'
					className={showAll ? disabled : active}
					onClick={() => toggleShow(false)}
				>
					ACTIVE
				</button>
			</div>
			<div>
				<button
					id='clear'
					className='text-xs mr-3 text-red-500 focus:outline-none hover:underline'
					onClick={clearDisplay}
				>
					CLEAR
				</button>
			</div>
		</div>
	)
}

export default ControlPanel