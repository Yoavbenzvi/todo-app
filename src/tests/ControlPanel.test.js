import { shallow } from "enzyme";
import React from "react";
import ControlPanel from "../components/ControlPanel.js";

describe('ControlPanel component', () => {
	it("expect to render ControlPanel component", () => {

		const wrapper = shallow(<ControlPanel />)
		expect(wrapper).toMatchSnapshot();
	});

	it("test click events", () => {
		const mockClick = jest.fn()
		const wrapper = shallow(<ControlPanel filterTasksArray={mockClick} clearDisplay={mockClick}/>)

		wrapper.find({id: 'all'}).simulate('click')
		wrapper.find({id: 'active'}).simulate('click')
		wrapper.find({id: 'clear'}).simulate('click')

		expect(mockClick).toHaveBeenCalledTimes(3)
	});
})