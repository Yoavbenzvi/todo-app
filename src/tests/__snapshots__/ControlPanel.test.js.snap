// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel component expect to render ControlPanel component 1`] = `
<div
  className="flex py-4 border-t border-gray-900 justify-between"
>
  <div>
    <button
      className="text-xs mr-3 hover:underline text-gray-500 focus:outline-none"
      id="all"
      onClick={[Function]}
    >
      ALL
    </button>
    <button
      className="text-xs mr-3 hover:underline text-gray-500 focus:outline-none text-green-500 font-bold"
      id="active"
      onClick={[Function]}
    >
      ACTIVE
    </button>
  </div>
  <div>
    <button
      className="text-xs mr-3 text-red-500 focus:outline-none hover:underline"
      id="clear"
    >
      CLEAR
    </button>
  </div>
</div>
`;
